zipline/data/loader.py

```diff
@@ -249,6 +249,7 @@ def ensure_benchmark_data(symbol, first_date, last_date, now, trading_day):
         data.to_csv(path)
     except (OSError, IOError, HTTPError):
         logger.exception('failed to cache the new benchmark returns')
+        raise
```

Comment:
Ah, I think I misread the intended logic here.  We only want to log and continue if it's `to_csv` that failed.  If the initial read fails, we don't have anything to return (hence the `UnboundLocalsError` that @jkleint ran into).